### DISCUSSION TITLE
Unsure how to debug `Unhandled Promise Rejection: RuntimeError: Out of bounds memory access` error

### DISCUSSION THREAD

**Discussion by bicarlsen** [wasm-bindgen/wasm-bindgen#3436] [Q&A]
I am creating a [Tauri](https://tauri.app/) app with a [Yew](https://yew.rs/) front end. I will sometimes receive a `Unhandled Promise Rejection: RuntimeError: Out of bounds memory access` error, but it seems to occur randomly. Sometimes it never occurs, sometimes it happens when starting the app up, other times when interacting with the app. So I haven't been able to find a reproducible pattern to incur it.

Does anybody have advice on how to go about debugging this?

**Comment by daxpedda** [comment]
That's the whole error?
It seems more like a warning to me, a `Promise` rejection that is unhandled doesn't necessary mean it's an error, could be a non-issue.

If it's an actual error or an exception thrown, you should easily be able to debug in the browser debugger. There is usually an option in the debugger to pause on exceptions.

**Comment by bicarlsen** [comment]
Yes, that was the only error. After playing around more, we've been able to narrow down what is causing the bug. I will keep the browser debugger in mind for the future.

Thank you
